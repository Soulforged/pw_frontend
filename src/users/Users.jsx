//@flow
import React from "react";
import { withHandlers } from "recompose";
import serialize from "form-serialize";
import UsersList from "./containers/UsersListContainer";
import UserDetails from "./containers/UserDetailsContainer";

const MainComponent = ({ filter, create }: { filter: () => void, create: () => void }) => (
  <div id="user-div">
    <div id="search-pnl" className="text-left">
      <div id="filter-row">
        <form onSubmit={filter} className="row">
          <div className="col-md-3 col-sm-3">
            <label htmlFor="criteria">
              Search criteria
              <select id="criteria" className="form-control" name="criteria" defaultValue="userName">
                <option value="userName">User Name</option>
                <option value="businessunit">Business Unit</option>
              </select>
            </label>
          </div>
          <div className="col-md-4 col-sm-4 txt-div">
            <label htmlFor="filter" className="lbl">
              Search text
              <input name="filter" className="form-control" placeholder="Text to search" />
            </label>
          </div>
          <div className="col-md-4 col-sm-4 slct-div hidden">
            <label htmlFor="srch-bunit" className="lbl">
              Business unit
              <select className="form-control" name="srch-bunit" />
            </label>
          </div>
          <div className="col-md-1 col-sm-1 text-right">
            <button id="filter-btn" className="btn" type="submit">GO</button>
          </div>
        </form>
      </div>
    </div>

    <div id="main-pnl" className="pnls">
      <div id="trans-wrap">
        <h4 className="form-header trebuchet bold">Back Office User(s) List
          <button className="add-new-btn pointer pull-right bold" onClick={create}>
            <i className="fa fa-plus theme" />
            New User
          </button>
        </h4>

        <div className="tbl-wrapper">
          <UsersList />
        </div>

        <UserDetails />
      </div>
    </div>
  </div>
);

export default withHandlers({
  filter: props => (event) => {
    event.preventDefault();
    const params = serialize(event.target, { hash: true });
    props.fetchUserByCriteria(params);
  },
  create: ({ openForm }) => () => openForm()
})(MainComponent);
