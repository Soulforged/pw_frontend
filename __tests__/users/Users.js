//@flow
import React from "react";
import { shallow } from "enzyme";
import Users from "src/users/Users";

it("matches snapshot", () => {
  const render = shallow(<Users />).dive();
  expect(render).toMatchSnapshot();
});
