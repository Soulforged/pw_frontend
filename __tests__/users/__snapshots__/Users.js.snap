// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <Filter />
  <div
    className="pnls"
    id="main-pnl"
  >
    <div>
      <h4
        className="form-header trebuchet bold"
      >
        Back office users
        <button
          className="add-new-btn pointer pull-right bold"
        >
          <i
            className="fa fa-plus theme"
          />
          New user
        </button>
      </h4>
      <div
        className="tbl-wrapper"
      >
        <lifecycle(branch(Component))
          columns={
            Array [
              Object {
                "Header": "ID",
                "accessor": "id",
                "maxWidth": 50,
              },
              Object {
                "Header": "First name",
                "accessor": "firstName",
              },
              Object {
                "Header": "Last name",
                "accessor": "lastName",
              },
              Object {
                "Header": "Username",
                "accessor": "userName",
              },
              Object {
                "Header": "Mobile number",
                "accessor": "mobileNumber",
              },
              Object {
                "Header": "Primary email",
                "accessor": "primaryEmail",
              },
              Object {
                "Header": "Business unit",
                "accessor": "businessUnitName",
              },
              Object {
                "Header": "Role",
                "accessor": "roleName",
              },
              Object {
                "Cell": [Function],
                "Header": "Status",
                "accessor": "status",
              },
            ]
          }
          createButtonTitle="New user"
          filter={<Filter />}
          title="Back office users"
        />
      </div>
    </div>
  </div>
</div>
`;
